Donation-platform/src/main/java/com/google/FindTrendingCharities.java

```diff
@@ -94,37 +103,180 @@ private double calcCharityTrendingScore(Charity charity) {
             avgReview = userRating;
         }
         //TODO: Integrate db with correct method to retreive tags for one charity
-        Collection<String> tags = charity.getTags();
-        double charityTagsScore = getTagTrendingScore(tags);
+        Collection<Tag> tags = new ArrayList<>();
+        try {
+           tags = db.getTagObjectsByIds(charity.getCategories());
+        }
+        catch (Exception e) {}
+        double charityTagsScore = 0;
+        try {
+           charityTagsScore = getTagTrendingScore(tags);
+        }
+        catch (Exception e) {}
         double charityTrendingScore = charityTagsWeight * charityTagsScore + avgReviewWeight * avgReview;
         return charityTrendingScore;
     }
 
     //TODO: Integrate db with correct method to retreive navRating for one charity
     private double calcCharityNavRating(Charity charity) {
-        double charityNavRating = charity.getCharityNavRating();
-        return charityNavRating * charityNavToScale;
+        //double charityNavRating;
+        //return charityNavRating * charityNavToScale;
+        return 0;
     }
 
     private boolean hasCharityNavRating(Charity charity) {
-        return true;
+        return false;
     }
 
     //return the average trending score of a collection of tags
-    private double getTagTrendingScore(Collection<String> tags) {
+    private double getTagTrendingScore(Collection<Tag> tags) throws Exception {
         double sumScores = 0;
         int numTags = tags.size();
-        for (String tag: tags) {
-            double tagScore = HardCodedCharitiesAndTags.tagScores.get(tag);
+        for (Tag tag: tags) {
+            //double tagScore = HardCodedCharitiesAndTags.tagScores.get(tag);
             //TODO: Integrate db with correct method to retreive trending score of specific tag
-            //double tagScore = tag.getTrendingScore();
+            double tagScore = tag.getTrendingScoreTag();
             
             //TODO: Use GoogleTrends API to update tag trending score
             //tagScore = use Google Trend API to get trending score
             sumScores += tagScore;
         }
-        //TODO: Update db with new trendingScore for charity
-
         return (sumScores / numTags);
     }
+
+    //add hardcoded charities into the db
+    private void dbSetUp() {
```

Comment:
This function would only ever have to be called once, right? Just want to make sure I don't need to call it in order to load the charities on the personalization page.

Reply:
Yes, I've already called it so the DB is set up with the charities already. If you call it again, make sure you clear the memcache in the datastore otherwise you'll get an error about too many queries being prepared.

Reply:
Such things we should move to a base class and extend for both personalized and trending. Please do that.

Reply:
done. extracted this to DBSetUpUtils.java